Cloning-Genitalia-Fix

ktccd opened the pull request:
:cl: Cloned Genitals Fix
fix: Various methods of cloning now gives the organs the clone's DNA says they should have and updates genital appearance along with clone updating of appearance.
/:cl:

ktccd committed:
Cloning-Genitalia-Fix

Fixes the issue with genitals not being re-created with cloning. This
should make it so other methods of coming back also gives the organs
your DNA says you should have, such as podding people.

TalkingCactus commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -50,6 +50,7 @@
 
 	internal_organs += new dna.species.mutanteyes()
 	internal_organs += new /obj/item/organ/brain
+	give_genitals()
```
`if(!(NOGENITALS in dna.species.species_traits))`

ktccd replied:
I'm... uhm, not sure what this means, I'm not very used to github. I don't really understand what you want changed :S.

ktccd replied:
Ok, so reading up a bit... this seems like, if anything, it should be in the give_genitals() proc, rather than hardcoded in this specific place. Nothing should ever get genitals if it has NOGENITALS trait in its dna and anything that has genitals needs to have dna to begin with, right? 'cause it's a carbon/human proc?

TalkingCactus replied:
It checks it as a failsafe, but doing one check here instead of potentially doing it several times later is better.

ktccd replied:
I already added a check in give_genitals now though? It's not... like... better to check outside it. The proc responsible for handling genitals should be doing the checks, right?

TalkingCactus replied:
That works, yes. (I was referring to each individual proc inside `give_genitals()`)

ktccd committed:
Merge remote-tracking branch 'refs/remotes/Citadel-Station-13/master' into Cloning-Genitalia-Fix

ktccd committed:
Sanity checks and fixes

Removed the OLD sanity check for drawing eyes. It was refusing to draw
eyes if you had ANY disability. 's far as I know, blind people still
have eyes and so do mute people.
Also, put a sanity check in give_genitals, just in case a species has
DNA with genitals but is a NOGENITALS species. Somehow. IT COULD HAPPEN,
ALRIGHT?! ^^

ktccd reviewed (commented):
IDK how the request changes thing works, but I implemented the suggestion in my latest commit...

ktccd committed:
TRAVIS LOVE ME AGAIN

This commit fixes the problem that made Travis hate me previously. Even
though THAT part of the code had not been changed...
Anyway, it looks more neat now, so that's good.
